src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    /// <summary>
+    /// CA1071: Constructor parameters should match property and field names
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class ConstructorParametersShouldMatchPropertyAndFieldNamesAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA1071";
+        internal const string ReferencedFieldOrPropertyName = "ReferencedFieldOrPropertyName";
+        internal const string DiagnosticReason = "DiagnosticReason";
```

Comment:
nits:

```suggestion
        internal const string ReferencedFieldOrPropertyName = nameof(ReferencedFieldOrPropertyName);
        internal const string DiagnosticReason = nameof(DiagnosticReason);
```

Reply:
I've added "Key" postfix to those names (but not to the values). Please, let me know if you still insist on using `nameof`.